src/main/kotlin/io/github/inductiveautomation/kindling/MainPanel.kt

```diff
@@ -330,22 +367,59 @@ class MainPanel : JPanel(MigLayout("ins 6, fill")) {
             }
         }
 
-        private fun macOsSetup(openFilesHandler: OpenFilesHandler) = Desktop.getDesktop().run {
-            // MacOS specific stuff
-            if (isSupported(Desktop.Action.APP_SUDDEN_TERMINATION)) {
-                disableSuddenTermination()
-            }
-            if (isSupported(Desktop.Action.APP_QUIT_STRATEGY)) {
-                setQuitStrategy(QuitStrategy.CLOSE_ALL_WINDOWS)
-            }
-            if (isSupported(Desktop.Action.APP_OPEN_FILE)) {
-                setOpenFileHandler(openFilesHandler)
+        private fun MainPanel.macOsSetup() {
+            if (Taskbar.isTaskbarSupported()) {
+                Taskbar.getTaskbar().apply {
+                    if (isSupported(Taskbar.Feature.ICON_IMAGE)) {
+                        setIconImage(Kindling.frameIcons.last())
+                    }
+                    if (isSupported(Taskbar.Feature.MENU)) {
```

Comment:
This is fun - adds the menu items to the dock right click menu:
<img width="397" alt="image" src="https://github.com/inductiveautomation/kindling/assets/39345262/31d8083d-d115-4ce9-8746-bb181a3f4776">